common/src/main/java/com/microsoft/identity/common/internal/broker/BrokerRequest.java

```diff
@@ -53,6 +54,7 @@
         final static String CLIENT_APP_NAME = "client_app_name";
         final static String CLIENT_APP_VERSION = "client_app_version";
         final static String CLIENT_VERSION = "client_version";
+        final static String CLIENT_SDKTYPE = "client_sdktype";
```

Comment:
Do we actually need to send this to the Broker? IIRC, MSAL / ADAL we don't need to send it based on the current design around adapters. Perhaps it is need for CPP?

Reply:
Yes it is need for CPP.

Reply:
It wasn't sent initially because Android.MSAL value was being hardcoded on the receiving end.

Reply:
Can you share CPP model for sending/unpacking broker request? Are they not using (custom) adapters?

Reply:
@shahzaibj  CPP will need to set SdkType as MSALCPP [here ]( https://github.com/AzureAD/microsoft-authentication-library-for-cpp/blob/develop/source/android/msal/app/src/main/java/com/microsoft/identity/internal/broker/BrokerRequestConverter.java#L79)and version too. Tanmay should make the changes there after this.

broker request will send this to broker (it will be MSAL or MSALCPP now )

Reply:
@kreedula I'm looking for the code where Broker will unpack the BrokerRequest that was sent to it via MSALCPP.

In broker, we currently have the `IBrokerRequestAdapter`, and we have implementations of it for ADAL/MSAL, but I don't see one for MSALCPP. Which adapter does MSALCPP use?

Reply:
it's the same format , i.e. common sends the request to broker . MSAL CPP just uses `BrokerMSALController" to call `acquireToken` or `acquireTokenSilent`